lib/Mojo/WebSocketProxy/Backend/JobAsync.pm

```diff
@@ -91,64 +145,63 @@ sub call_rpc {
     my $method = $req_storage->{method};
     my $msg_type = $req_storage->{msg_type} ||= $req_storage->{method};
 
-    # We'd like to provide some flexibility for people trying to integrate this into
-    # other systems, so any combination of Job::Async::Client, Job::Async and/or IO::Async::Loop
-    # instance can be provided here.
-    $self->{client} //= do {
-        # We don't hold a ref to this, since that might introduce unfortunate cycles
-        $self->{loop} //= do {
-            require IO::Async::Loop::Mojo;
-            local $ENV{IO_ASYNC_LOOP} = 'IO::Async::Loop::Mojo';
-            IO::Async::Loop->new;
-        };
-        $self->{loop}->add(my $jobman = Job::Async->new);
-
-        # Let's not pull it in unless we have it already, but we do want to avoid sharing number
-        # sequences in forked workers.
-        Math::Random::Secure::srand() if Math::Random::Secure->can('srand');
-        my $client_job = $jobman->client(redis => $self->{redis});
-        $client_job->start->retain;
-        $client_job;
-    };
-
     $req_storage->{call_params} ||= {};
     my $rpc_response_cb = $self->get_rpc_response_cb($c, $req_storage);
 
-    my $before_get_rpc_response_hook = delete($req_storage->{before_get_rpc_response}) || [];
-    my $after_got_rpc_response_hook  = delete($req_storage->{after_got_rpc_response})  || [];
-    my $before_call_hook             = delete($req_storage->{before_call})             || [];
+    my $before_get_rpc_response_hooks = delete($req_storage->{before_get_rpc_response}) || [];
+    my $after_got_rpc_response_hooks  = delete($req_storage->{after_got_rpc_response})  || [];
+    my $before_call_hooks             = delete($req_storage->{before_call})             || [];
+    my $rpc_failure_cb                = delete($req_storage->{rpc_failure_cb}) || 0;
+
     my $params = $self->make_call_params($c, $req_storage);
     $log->debugf("method %s has params = %s", $method, $params);
-    $_->($c, $req_storage) for @$before_call_hook;
-    $self->client->submit(
-        name   => $req_storage->{name},
-        params => encode_json_utf8($params)
-        )->on_ready(
-        sub {
+
+    foreach my $hook (@$before_call_hooks) { $hook->($c, $req_storage) };
+
+    Future->wait_any(
+        $self->client->submit(
+            name   => $req_storage->{name},
+            params => encode_json_utf8($params)
+        ),
+        $self->loop->timeout_future(after => QUEUE_TIMEOUT)
+        )->on_ready(sub {
             my ($f) = @_;
             $log->debugf('->submit completion: ', $f->state);
 
-            $_->($c, $req_storage) for @$before_get_rpc_response_hook;
+            foreach my $hook (@$before_get_rpc_response_hooks) { $hook->($c, $req_storage) };
```

Comment:
i think it was fine how it was. as in there is no need for this change

Reply:
@nael-binary There is nothing wrong with using `$_` here, but since there are a number of constructs around setting the default variable in different ways, I've accepted @michaelmueller-binary 's suggestion to use named variables for more readability.